### ISSUE TITLE
Spam Classification

### ISSUE THREAD

**Issue by G3024** [G3024/Geo#4]
import pandas as pd
df = pd.read_csv("spam1.csv")
df.groupby('Category').describe()
df['spam']=df['Category'].apply(lambda x: 1 if x=='spam' else 0)

from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(df.Message,df.spam)

from sklearn.feature_extraction.text import CountVectorizer
v = CountVectorizer()
X_train_count = v.fit_transform(X_train.values)
X_train_count.toarray()[:2]

from sklearn.naive_bayes import MultinomialNB
model = MultinomialNB()
model.fit(X_train_count,y_train)    

emails = [
    'Hey mohan, can we get together to watch footbal game tomorrow?',
    'Upto 20% discount on parking, exclusive offer just for you. Dont miss this reward!'
]
emails_count = v.transform(emails)
model.predict(emails_count) 

X_test_count = v.transform(X_test)
model.score(X_test_count, y_test) #0.9849246231155779

from sklearn.pipeline import Pipeline
clf = Pipeline([('vecetorize', CountVectorizer()),('nb', MultinomialNB())])
clf.fit(X_train, y_train)
print(clf.predict(X_test)) #[1 0 0 ... 0 1 0]
print(clf.score(X_test, y_test)) #0.9892318736539842
